### PR TITLE
introduce actions/cache

### PR DESCRIPTION
**Pull request by sakamoto-souta**
# Summary
Introduce [actions/cache](https://github.com/actions/cache)
# Purpose
To reduce build time
# Contents
Introduce [actions/cache](https://github.com/actions/cache) for building images.
This PR reduces build time by using cache.

[Before](https://github.com/acompany-develop/QuickMPC/actions/runs/4922134220/jobs/8792626303) -> [After](https://github.com/acompany-develop/QuickMPC/actions/runs/4931640009)
small_test_cc : 18m11s -> 17m14s
small_test_mc : 3m34s -> 3m11s
small_test_bts : 5m6s -> 4m45s
medium_test_cc : 22m22s -> 9m27s
medium_test_mc : 2m30s -> 3m14s
medium_test_libclient : 27m32s -> 12m43s
**medium_test_container : 47m32s -> 33m56s**

Note that these results are for a case where the code has not been modified and the cache is 100% active.
In particular, since it is currently implemented with mode=min to avoid getting caught in the 10GB limit of the cache, a small change may cause the cache to become ineffective.
# Testing Methods Performed

### CODE REVIEW

**Comment by come2ry** [comment]
@sakamoto-souta
I saw a speedup of about 10 minutes!
Can you comment on what the impact was before and after this PR, in numerical terms?

image
image

File: .github/workflows/test-server-all.yml [hunk 1]
```diff
@@ -50,14 +50,38 @@ jobs:
...
 
-      - name: Build images and run computation_container
-        run: make test t=./computation_container m=build p=small
-        working-directory: ./scripts
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-small_cc-${{ hashFiles('./packages/server/computation_container/**') }}
+          restore-keys: ${{ runner.os }}-small_cc-
```
Comment (come2ry) [inline .github/workflows/test-server-all.yml:58]: @sakamoto-souta 
I feel uncomfortable with the suffix ending in -.
Is there an intention behind this naming?

Comment (sakamoto-souta) [reply .github/workflows/test-server-all.yml:58]: I just imitated the [official example](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#example-using-multiple-restore-keys).
I don't have any intention.

**Review by come2ry — APPROVED** [review]
LGTM

File: .dockerignore [hunk 2]
```diff
@@ -1,2 +1,2 @@
 **/vendor
-**/.cache
+**/.cache
```
Comment (arukuka) [inline .dockerignore:2]: Please add new line at EOF.

```suggestion
**/.cache

```

**Review by arukuka — CHANGES_REQUESTED** [review]
Thank you for your contribution! :pray:  
Sorry for the small details, but please check the comments.

File: .dockerignore (hunk 2 above)
Comment (sakamoto-souta) [reply .dockerignore]: [8549f25](https://github.com/acompany-develop/QuickMPC/pull/180/commits/8549f258fea615355eb12f30dd452311cb965133)

Comment (arukuka) [reply .dockerignore]: Thank you!

**Pull request merged**